// Copyright 2010 Google Inc. All Rights Reserved.
// Author: jmarantz@google.com (Joshua Marantz)

// Protobufs to help encode rewritten resources as URLs.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package net_instaweb;

message ResourceUrl {
  optional string origin_url = 1 [default = "" ];
};

message CssUrl {
  optional string origin_url = 1 [default = "" ];
  optional string media = 2 [default = "" ];
};

message CssCombineUrl {
  repeated CssUrl element = 1;
};

message ImgRewriteUrl {
  required string origin_url = 1;
  optional int32 width = 2 [default = -1];
  optional int32 height = 3 [default = -1];
};
